task_processing/plugins/mesos/timeout_executor.py

```diff
@@ -83,11 +96,7 @@ def run(self, task_config):
             deadline=task_config.timeout + time.time()
         )
         with self.tasks_lock:
-            for idx, entry in enumerate(self.running_tasks):
-                if new_entry.deadline <= entry.deadline:
-                    self.running_tasks.insert(idx, new_entry)
-                    return
```

Comment:
My initial thought was that we wanted to run no matter what, but looking over it again I see that it's to prevent re-running duplicated tasks. I'll fix it to return and prevent run() from executing, like before.